### PR TITLE
MF-111 Login page should skip location select if only one location is available

### PR DESCRIPTION
**Pull request by brandones**
The general approach is to have the root component fetch loginLocations and pass them to the Login and ChooseLocation components.

### CODE REVIEW

File: src/login/login.component.test.tsx [hunk 1]
```diff
@@ -50,7 +58,7 @@ describe(`<Login />`, () => {
     expect(performLogin).toHaveBeenCalled();
   });
 
-  it(`send the user to the location select page on login`, () => {
+  xit(`send the user to the location select page on login if there is more than one location`, () => {
```
Comment (brandones) [inline src/login/login.component.test.tsx]: The `historyMock` is colliding between this test and the new one. With this test disabled, everything works. When this test is enabled, `historyMock.push` in the new test seems to receive the value it has in this test.

Comment (fatmali) [reply src/login/login.component.test.tsx]: take a look at how I mocked this in [patient search](https://github.com/openmrs/openmrs-esm-home/blob/master/src/helpers/render-with-router.tsx), might help out.

Comment (brandones) [reply src/login/login.component.test.tsx]: Thanks @fatmali ! Opened https://github.com/openmrs/openmrs-esm-login/pull/40

Comment (fatmali) [reply src/login/login.component.test.tsx]: @brandones should we close this as #40 has been merged?

Comment (brandones) [reply src/login/login.component.test.tsx]: @fatmali No, now that #40 has been merged, I can continue working on this one. Thank you for your support :)

**Pull request merged**